pkg/engine/anchor/anchormap.go

```diff
@@ -0,0 +1,44 @@
+package anchor
+
+// AnchorMap - contains map of anchors
+type AnchorMap struct {
+	// anchorMap - for each anchor key in the patterns it will maintain information if the key exists in the resource
+	// if anchor key of the pattern exists in the resource then (key)=true else (key)=false
+	anchorMap map[string]bool
+	// AnchorError - used in validate to break execution of the recursion when if condition fails
+	AnchorError validateAnchorError
+}
+
+// NewAnchorMap -initialize anchorMap
+func NewAnchorMap() *AnchorMap {
```

Comment:
Where is this used?

Reply:
In the engine validation...

Reply:
OK this was just a refactor, no wonder why I couldn't find any related change in this PR.

Reply:
Before that the struct was named `AnchorKey`... but it actually is a map 🤷